lib/programmemory.h

```diff
@@ -182,13 +180,13 @@ struct ProgramMemoryState {
     ProgramMemory get(const Token* tok, const Token* ctx, const ProgramMemory::Map& vars) const;
 };
 
-std::vector<ValueFlow::Value> execute(const Scope* scope, ProgramMemory& pm, const Settings* settings);
+std::vector<ValueFlow::Value> execute(const Scope* scope, ProgramMemory& pm, const Settings& settings);
 
 void execute(const Token* expr,
              ProgramMemory& programMemory,
              MathLib::bigint* result,
              bool* error,
-             const Settings* settings = nullptr);
```

Comment:
Cases where this parameter was being omitted need to be reviewed as it changes the behavior. I have more such changes in an additional PR and will add TODOs for the callers there.